test/logstream/kubelogs.go

```diff
@@ -49,54 +53,99 @@ var _ streamer = (*kubelogs)(nil)
 // timeFormat defines a simple timestamp with millisecond granularity
 const timeFormat = "15:04:05.000"
 
+func (k *kubelogs) startForPod(eg *errgroup.Group, pod *corev1.Pod) {
+	k.watchedPods.Insert(pod.Name)
+	// Grab data from all containers in the pods.  We need this in case
+	// an envoy sidecar is injected for mesh installs.  This should be
+	// equivalent to --all-containers.
+	for _, container := range pod.Spec.Containers {
+		// Required for capture below.
+		pod, container := pod, container
+		eg.Go(func() error {
+			options := &corev1.PodLogOptions{
+				Container: container.Name,
+				// Follow directs the api server to continuously stream logs back.
+				Follow: true,
+				// Only return new logs (this value is being used for "epsilon").
+				SinceSeconds: ptr.Int64(1),
+			}
+
+			req := k.kc.Kube.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, options)
+			stream, err := req.Stream()
+			if err != nil {
+				return err
+			}
+			defer stream.Close()
+			// Read this container's stream.
+			scanner := bufio.NewScanner(stream)
+			for scanner.Scan() {
+				k.handleLine(scanner.Bytes())
+			}
+			// Pods get killed with chaos duck, so logs might end
+			// before the test does.
+		})
+	}
+}
+
+func (k *kubelogs) watchPods(t test.TLegacy, eg *errgroup.Group) {
+	wi, err := k.kc.Kube.CoreV1().Pods(k.namespace).Watch(metav1.ListOptions{})
+	if err != nil {
+		t.Logf("Logstream knative pod watch failed, logs might be missing: %v", err)
```

Comment:
Should we surface this error similar to the `List` error in `init`?

Reply:
I debated that... I am not sure failing a test, because of logging failure is a valid concern?

Reply:
But I'll go along with what we have now.